utils/threedi_result_aggregation/threedigrid_ogr.py

```diff
@@ -1,209 +1,69 @@
-
-    all_attributes = default_attributes
-    all_attributes.update(attributes)
-    all_attr_data_types = default_attr_types
-    all_attr_data_types.update(attr_data_types)
-
-    # create output layer
-    srs = osr.SpatialReference()
-    srs.ImportFromEPSG(int(threedigrid_src.epsg_code))
-    out_layer = tgt_ds.CreateLayer(
-        out_layer_name, srs, geom_type=out_geom_type
-    )
-
-    # create fields
-    for attr in all_attributes.keys():
-        field = ogr.FieldDefn(attr, all_attr_data_types[attr])
-        out_layer.CreateField(field)
-
-    feature_defn = out_layer.GetLayerDefn()
-
-    # create features
-    for i in range(threedigrid_src.count):
-        if (not np.all(np.isfinite(coords[:, i]))) or (
-            not np.all(coords[:, i] != -9999)
-        ):  # skip if coordinates are invalid
-            continue
-        else:
-            # create feature geometry
-            feature = ogr.Feature(feature_defn)
-            geom = ogr.Geometry(out_geom_type)
-            if src_type == Nodes:
-                x, y = coords[:, i]
-                geom.SetPoint(0, x, y)
-                feature.SetGeometry(geom)
-            elif src_type == Lines:
-                x0, y0, x1, y1 = coords[:, i]
-                geom.AddPoint(float(x0), float(y0))
-                geom.AddPoint(float(x1), float(y1))
-                feature.SetGeometry(geom)
-            elif src_type == Cells:
-                xmin, ymin, xmax, ymax = coords[:, i]
-                geom_ring = ogr.Geometry(ogr.wkbLinearRing)
-                geom_ring.AddPoint(xmin, ymin)
-                geom_ring.AddPoint(xmin, ymax)
-                geom_ring.AddPoint(xmax, ymax)
-                geom_ring.AddPoint(xmax, ymin)
-                geom_ring.AddPoint(xmin, ymin)
-                geom.AddGeometry(geom_ring)
-                if not geom.IsValid():
-                    continue
-                elif geom.IsEmpty():
-                    continue
-                else:
-                    feature.SetGeometry(geom)
-
-            # create feature attributes
-            for attr in all_attributes.keys():
-                val = all_attributes[attr][i]
-                if all_attr_data_types[attr] in [ogr.OFTInteger]:
-                    if val is not None:
-                        val = int(val)
-                elif all_attr_data_types[attr] in [ogr.OFTString]:
-                    if isinstance(val, bytes):
-                        val = val.decode("utf-8")
-                    else:
-                        val = str(val)
-                elif all_attr_data_types[attr] in [ogr.OFTReal]:
-                    if np.isnan(val):
-                        val = None
-                if val is not None:
-                    feature[attr] = val
-
-            # create the actual feature
-            out_layer.CreateFeature(feature)
+    # open gridadmin.gpkg as input datasource
+    src_ds = ogr.Open(gridadmin_gpkg, 1)
```

Comment:
I think the second parameter is the open mode, where 1 is "write"? Default is read (which you need here), so I think this can be just ``src_ds = ogr.Open(gridadmin_gpkg)